bears/general/InvalidLinkBear.py

```diff
@@ -43,8 +43,23 @@ def get_status_code(url, timeout):
     def find_links_in_file(file, timeout, link_ignore_regex):
         link_ignore_regex = re.compile(link_ignore_regex)
         regex = re.compile(
-            r'(https?://[^.:%\s_/?#[\]@\\]+\.(?:[^\s()%\'"`<>|\\]+|'
-            r'\([^\s()%\'"`<>|\\]*\))*)(?<!\.)(?<!,)')
+            r"""
+            (https?://                  # http:// or https:// as only these
+                                        # 2 are supported by the ``requests``
+                                        # library
+            [^.:%\s_/?#[\]@\\]+         # Initial Part of Domain
+            \.                          # A required dot (.)
+            (
+                (?:[^\s()%\'"`<>|\\]+)  # Path Name
+            |                           # OR
+                \([^\s()%\'"`<>|\\]*\)  # Path Name contained within ()
```

Comment:
Aah yes, I now see it clearly, Thank you for pointing it out. Also, I think this has been done because both the parts of the ``OR`` include ``[^\s()%\'"`<>|\\]`` wherein ``(`` and ``)`` are also present. This makes the regex ``not`` match the URLs containing unbalanced round brackets. And thus the second part of ``OR`` is required to match ``path names within ()``. So, a quick question, 
1) Should I document it as you stated ?
2) Or Should I actually remove ``(`` and ``)`` from ``[^\s()%\'"`<>|\\]``. By doing this, we won't need the second part of ``OR`` and also, maybe the unbalanced round brackets can also be matched.